src/import-data/useImportRequest.ts

```diff
@@ -70,11 +71,36 @@ const getFileImportRequest = (queryParams: QueryParams, type: FileImportRequest[
   return { type, url };
 };
 
+/**
+ * Generate a snapshot manifest on demand for importing a TDR snapshot.
+ */
+const generateSnapshotManifest = async (snapshotId: string): Promise<URL> => {
+  const { id } = await DataRepo().snapshot(snapshotId).exportSnapshot();
+  const jobApi = Ajax().DataRepo.job(id);
+  let jobInfo = await jobApi.details();
+  while (jobInfo.job_status === 'running') {
+    // sleep for 1000 ms
+    await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
FYI, there's a `delay` function in `@terra-ui-packages/core-utils` to do this.

```suggestion
    await delay(1000);
```